### DISCUSSION TITLE
Repo Setup: [next-auth][error][NO_SECRET]

### DISCUSSION THREAD

**Discussion by 47seconds** [lukevella/rallly#1188] [Q&A]
On a fresh setup, i forked and cloned the repo, and followed instructions in the readme, and made a .env file copied from .env.development, as well as set up docker and every thing seemed fine.

The problem is when I run command 'yarn dev' and go to specified port where project is running, i.e. 3000, page won't load and the terminal console in IDE shows this:

│  ✓ Compiled /api/auth/[...nextauth] in 333ms (241 modules)
│  GET /api/auth/error?error=Configuration 302 in 973ms
│ [next-auth][error][NO_SECRET] 
│ https://next-auth.js.org/errors#no_secret

I tried console.log(process.env.SECRET_PASSWORD) in [...nextauth].ts to check if there is some problem loading these variables but it was able to print these:

│  ✓ Compiled /api/auth/[...nextauth] in 333ms (241 modules)
│ abcdef1234567890abcdef1234567890
│  GET /api/auth/error?error=Configuration 302 in 973ms
│ [next-auth][error][NO_SECRET] 
│ https://next-auth.js.org/errors#no_secret

I also tried changing the variable name to NEXTAUTH_SECRET everywhere in project, but it still gave me same error.
What am I missing? or am I doing something wrong? This is going to be my first Open source working project, please help!

**Comment by lukevella** [comment]
Are you try to self-host or are you trying to get a development environment working?

**Reply by 47seconds** [reply]
I am trying to set the development environment to start development and solve issues and add features/enhancements.

**Reply by qhantom** [reply]
@47seconds Just create an `.env` file inside `apps/web` and put the `NEXTAUTH_SECRET` variable inside it with some characters as values. @lukevella For some reason, Next.js does not look after the `.env` file when it's a few levels up from the web-package. How is your local setup with this?

BTW: Great application! Much appreciation here 🙏

**Reply by lukevella** [reply]
I have my `.env` file in the root of the project. If you run `yarn dev` from the root it should load the env vars from it. 

`NEXTAUTH_SECRET` isn't used. We use `SECRET_PASSWORD` instead to maintain backwards compatibility with older versions that pre-date next-auth being used in the project.

**Reply by qhantom** [reply]
I did that but get the error `[next-auth][error][NO_SECRET]`. 🤔

**Reply by lukevella** [reply]
Not sure man. I can't do 1:1 support like this cause I'd never get anything done. If you figure this out, let me know if you find an issue with the repo.

**Reply by qhantom** [reply]
Wasn't my expectation ✌️ Will do!

**Reply by lukevella** [reply]
Hey, I recently set up the repo from scratch and saw the same error. I think the reason was because I hadn't run `yarn db:reset` which sets up the db schema and runs the seed script. `yarn db:deploy` also works if you don't want the seed data. You may want to try that and see if it fixes the issue.
